crates/matrix-sdk/src/client/futures.rs

```diff
@@ -94,6 +96,9 @@ where
                     return res;
                 }
 
+                let refresh_token =
+                    client.session().clone().and_then(|s| s.refresh_token().clone());
```

Comment:
If I'm not mistaken, this should work and avoid a full clone of the session:

```suggestion
                    client.session().as_ref().and_then(|s| s.refresh_token().clone());
```